database/postgres/postgres_collector.go

```diff
@@ -0,0 +1,294 @@
+package postgres
+
+import (
+	"github.com/frain-dev/convoy/pkg/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"strings"
+	"time"
+)
+
+// Namespace used in fully-qualified metrics names.
+const namespace = "convoy"
+
+const delaySeconds = 1
```

Comment:
@mekilis, the sample time should be exposed as an instance-level configuration.

Reply:
@mekilis, because this can have a significant performance impact, we should add an instance-level configuration to enable/disable metrics. Something like:

```json
{
  "metrics": {
    "backend": "prometheus",
    "prometheus": {
      // sample time in seconds.
      "sample_time": 10 
    }
  }
}
```

Reply:
Got it.

Reply:
Done